app/scripts/modules/core/src/serverGroup/ServerGroupHeader.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import { get } from 'lodash';
+
+import { NgReact, ReactInjector } from 'core/reactShims';
+import { Application } from 'core/application';
+import { IServerGroup } from 'core/domain';
+import { JenkinsViewModel } from 'core/serverGroup/ServerGroup';
+import { EntityNotifications } from 'core/entityTag/notifications/EntityNotifications';
+import { HealthCounts } from 'core/healthCounts';
+import { CloudProviderLogo } from 'core/cloudProvider';
+import { LoadBalancersTagWrapper } from 'core/loadBalancer';
+import { ServerGroupManagerTag } from 'core/serverGroupManager/ServerGroupManagerTag';
+import { ISortFilter } from 'core/filterModel';
+import { Overridable } from 'core/overrideRegistry';
+
+export interface IServerGroupHeaderProps {
+  application: Application;
+  images?: string;
+  isMultiSelected: boolean;
+  jenkins: JenkinsViewModel;
+  serverGroup: IServerGroup;
+  sortFilter: ISortFilter;
+}
+
+export class LoadBalancers extends React.Component<IServerGroupHeaderProps> {
```

Comment:
Made all these things full fledged components.  Opinions welcomed

Reply:
I'm guessing you went with classes because some of these are overridden and right now `Overridable` is only a class decorator?

If that's the case, I'd say it'd be very nice to have an equivalent decorator for SFCs that is just a normal wrapper — not sure what you'd want the signature to look like, but could be `overridable('namespace.ComponentName', () => ...)`, `overridable('namespace.ComponentName')(() => ...)`, etc.

`connect()` et al. does this in redux (i.e. there is also a `@Connect()`) and I think it's probably going to get more than a bit tedious to require that every overridable component is a full-on class — even this file on its own has a lot of visual noise.

Reply:
I actually prefer class based components over SFC for things I might want to inspect in the React Dev Tools (which doesn't play nicely with SFC).  Originally I marked all these class components as `@Overridable`, but that seemed overkill since I'm only actually overriding the one component.

We do have an HOC (which the `@Overridable` decorator itself uses):  `overridableComponent(() => <h1>hey</h1>, "some.key")`

https://github.com/spinnaker/deck/blob/b1c112ac8f3ab253bfe760d9b1adf2efdaeab82a/app/scripts/modules/core/src/overrideRegistry/Overridable.tsx#L44-L51

Reply:
oh duh, I forgot the decorator delegated down to `overridableComponent` 🤦‍♂️ 

Can you say more about what is difficult with devtools? At present I'm only familiar with the unknown component name issue, which afaik is a solved problem now (within the last ~6m-1yr) for both babel and typescript. Are there other things that react-devtools doesn't do well with stateless components?

I'd love to dig into anything that's a tooling issue and could be solved, but to be clear I'm totally on-board with classes if there's nothing we can do to fix the tooling currently.

Reply:
Gross, haven't run into that one before. Hopefully they figure out some way around the lack of an instance ref and we can stop worrying about SFC + tooling.